core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -106,6 +105,20 @@ abstract class TaskContext extends Serializable {
   def addOnCompleteCallback(f: () => Unit)
 
   /**
+   * Add a (Java friendly) listener to be executed on task interruption. We add this
+   * listener for some more clean works. An example use is to stop `receiver supervisor`
+   * properly.
+   */
+  def addTaskInterruptedListener(listener: TaskInterruptionListener): TaskContext
```

Comment:
For consistency, I think we should call this `addTaskInterruptionListener`, since the actual listener class is named `TaskInterruptionListener` and for consistency with the other listener, which is named "completion" rather than "complete".